rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,1153 @@
+            ```
+
+            The encoding change for numbers and dates is important and can have a big impact on application behavior. If you have
+            a query that targets a field that was previously string-encoded, but is now encoded numerically, you must update the query. Numeric
+             fields must be searched with a NumericRangeQuery. You must also make sure that all fields targeted by faceting are string encoded.
+            If you use the Search query DSL, the correct query should be created automatically for you.
+            ]]></message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscellaneous Changes in Hibernate Search 5.x" />
+                    <link href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/bridge/builtin/package-summary.html" title="org.hibernate.search.bridge.builtin package javadoc" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <!-- When using @Field(indexNullAs=) to encode a null marker value in the index, -->
+        <!-- the type of the marker must be compatible with all other values that are indexed in that same field. -->
+        <!-- For example, it was previously possible to encode a null value for numeric fields using a string "null". -->
+        <!-- This is no longer allowed. Instead, you must choose a number to represent the null value, such as -1. -->
+        <rule id="hsearch-00117">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.Field">
+                    <location>ANNOTATION</location>
+                    <annotation-literal name="indexNullAs" pattern="{*}" />
+                    <!-- https://issues.jboss.org/browse/WINDUP-974 -->
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Changes in indexing null values" effort="1" severity="mandatory">
+                    <message>When using `@Field(indexNullAs=)` to encode a null marker value in the index, the type of the marker must be compatible with all other values that are indexed in that same field.
+                    For example, it was previously possible to encode a null value for numeric fields using a string _null_.
+                    This is no longer allowed. Instead, you must choose a number to represent the null value, such as -1. </message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscellaneous Changes in Hibernate Search 5.x" />
```

Comment:
Could also link to http://hibernate.org/search/documentation/migrate/5.5/#null-value-tokens-for-numeric-fields